src/konfo_backend/suosikit/suosikit.clj

```diff
@@ -47,4 +47,40 @@
                                          (assoc :toteutusOid (get-in t [:oid]))
                                          (assoc :tutkintonimikkeet (get-in koulutukset [(:koulutusOid t) :metadata :tutkintonimike])))))))))
          (flatten)
-         (filter #(contains? hakukohde-oids (:hakukohdeOid %))))))
\ No newline at end of file
+         (filter #(contains? hakukohde-oids (:hakukohdeOid %))))))
+
+(defn into-map-by [get-key items]
+  (into {} (map #(vec [(get-key %) %]) items)))
+
+(defn get-vertailu-by-hakukohde-oids
+  [hakukohde-oids-seq]
+  (let [hakukohde-oids (set hakukohde-oids-seq)
+        hakukohteet (hakukohde/get-many hakukohde-oids)
+        toteutukset-by-oid (into-map-by :oid (toteutus/get-many (distinct (map :toteutusOid hakukohteet))))
+        oppilaitokset-res (oppilaitos/get-many (distinct (mapcat :oppilaitokset (vals toteutukset-by-oid))) false)
+        osat-by-oid (mapcat #(map (fn [osa] [(:oid osa) %]) (:osat %)) oppilaitokset-res)
+        oppilaitokset-by-oid (map (fn [oppilaitos] [(:oid oppilaitos) oppilaitos]) oppilaitokset-res)
+        orgs-by-oid (into {} (concat osat-by-oid oppilaitokset-by-oid))]
+    (map (fn [hk] (let [toteutus (get-in toteutukset-by-oid [(:toteutusOid hk)])
```

Comment:
Toteutuksesta taidetaan lukea tässä vain metadataa ylipäätään, joten jos haluaisi, voisi kai tässä lukea suoraan sen metadatan `toteutus-metadata` -muuttujaan, ja sitten myöhemmin henkeen `(:tyyppi toteutus-metadata)` jne. 

(Oppilaitokselle ja hakukohteelle ei näytä voivan tehdä samaa temppua kun dataa luetaan muualtakin rakenteesta - toki voi olla erikseen määritelty oppilaitos ja oppilaitos-metadata jos haluaa toistaa tuota get-in:iä vähemmän myöhemmin...)